src/pywifes/splice.py

```diff
@@ -0,0 +1,360 @@
+
+        # Run over every point (i,j) in the spatial plane
+        for i in range(y_dim):
+            for j in range(x_dim):
+                # Red
+                red_flux = red_flux_cube[:, i, j]
+                flux_R = np.array(AR * red_flux).ravel()
+
+                red_fluxvar = red_fluxvar_cube[:, i, j]
+                diag_R = sp.dia_matrix(
+                    ([red_fluxvar], [0]), shape=[red_NAXIS1, red_NAXIS1]
+                )
+                fluxvar_R = np.array((AR * diag_R * AR.T).sum(axis=1)).ravel()
+
+                # Blue
+                blue_flux = blue_flux_cube[:, i, j]
+                flux_B = np.array(AB * blue_flux).ravel()
+
+                blue_fluxvar = blue_fluxvar_cube[:, i, j]
+                diag_B = sp.dia_matrix(
+                    ([blue_fluxvar], [0]), shape=[blue_NAXIS1, blue_NAXIS1]
+                )
+                fluxvar_B = np.array((AB * diag_B * AB.T).sum(axis=1)).ravel()
+
+                BUFFER = 10.0
+
+                # Check for wavelength overlap
+                blue_only = np.where(wl <= red_min_wavelength + BUFFER)
+                overlap = np.where(
+                    (wl < blue_max_wavelength - BUFFER)
+                    & (wl > red_min_wavelength + BUFFER)
+                )
+                red_only = np.where(wl >= blue_max_wavelength - BUFFER)
+
+                # Average the two taking into account the buffer region and weighting
+                flux = np.zeros(len(flux_B), float)
+                fluxVar = np.zeros(len(flux_B), float)
+
+                flux[blue_only] = flux_B[blue_only]
+                fluxVar[blue_only] = fluxvar_B[blue_only]
+
+                fluxVar[overlap] = 1.0 / (
+                    1.0 / fluxvar_B[overlap] + 1.0 / fluxvar_R[overlap]
+                )
+                flux[overlap] = (
+                    flux_B[overlap] / fluxvar_B[overlap]
+                    + flux_R[overlap] / fluxvar_R[overlap]
+                ) * fluxVar[overlap]
+
+                flux[red_only] = flux_R[red_only]
+                fluxVar[red_only] = fluxvar_R[red_only]
+                flux_cube[:, i, j] = flux
+                fluxvar_cube[:, i, j] = fluxVar
+
+        # Reshape the cube to the corresponding shape
+        wave_dim = len(flux)
+        flux_cube = flux_cube.reshape(wave_dim, y_dim, x_dim)
+        fluxvar_cube = fluxvar_cube.reshape(wave_dim, y_dim, x_dim)
+
+        return flux_cube, fluxvar_cube
+
+
+def splice_cubes(blue_path, red_path, output_path):
+    """
+    Main routine
+    """
+
+    red_header = fits.getheader(red_path, 0)
+    blue_header = fits.getheader(blue_path, 0)
+
+    # Join the cubes
+    print("Splicing cubes.")
+    flux, fluxVar = join_cubes(blue_path, red_path)
+
+    if flux is None:
+        print("No spectral overlap between the cubes.")
+
+    else:
+
+        # Write out the results
+        # Use the header in red arm to start with
+        # Add additional blue CCD keywords as required
```

Comment:
Fullstops could be nice, hard to tell if it's 3 separate comments, or 1 multiline comment.

Reply:
Yeah, I put them all in a single line.